### PR TITLE
Refactor search and related stores to add other media types

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Related to #118 by @zackkrida 

## Description

This PR refactors the `related-images-store` to make it possible to add other media types. The file is renamed to `related-media-store`.

## Technical details

Several actions and mutations are duplicated now (have both `_MEDIA` and `_IMAGES` versions) to keep the scope of this PR small, but will be renamed in the next PR.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/pages/photos/_id.vue [hunk 1]
```diff
@@ -81,19 +85,24 @@ const PhotoDetailPage = {
...
       await Promise.all([
         store.dispatch(FETCH_IMAGE, { id: route.params.id }),
-        store.dispatch(FETCH_RELATED_IMAGES, { id: route.params.id }),
+        store.dispatch(FETCH_RELATED_MEDIA, {
+          mediaType: IMAGE,
+          id: route.params.id,
+        }),
       ])
     } catch (err) {
+      console.log('oops, ', err)
```
Comment (obulat) [inline src/pages/photos/_id.vue:105]:
```suggestion
```

File: src/store-modules/related-media-store.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      .catch((error) => {
+        dispatch(HANDLE_IMAGE_ERROR, error)
+      })
+  },
+})
+
+const mutations = {
+  /**
+   * @param _state
+   * @param {('image'|'audio')} mediaType
```
Comment (obulat) [inline src/store-modules/related-media-store.js:56]:
```suggestion
   * @param {'image'|'audio'} mediaType
```

**Review by krysal — APPROVED** [review]
This looks GREAT to me! And it'd be great if we can merge it before anything else to avoid being left behind in the changes.

**Review by zackkrida — APPROVED** [review]
good review timing @krysal and nice work @obulat

**Comment by zackkrida** [comment]
@krysal merged 😄

**Pull request merged**